sympy/solvers/ode.py

```diff
@@ -6475,10 +6475,11 @@ def _linear_2eq_order1_type1(x, y, t, r):
     .. math:: x = C_1 (bk t - 1) + b C_2 t , y = k^{2} b C_1 t + (b k^{2} t + 1) C_2
 
     """
-    l = Symbol('l')
+    l = Dummy()
     C1, C2, C3, C4 = symbols('C1:5')
-    l1 = RootOf(l**2 - (r['a']+r['d'])*l + r['a']*r['d'] - r['b']*r['c'], 0)
```

Comment:
Did you know that RootOf has a specific argument for unknown variable of Poly?  E.g. `RootOf(x**2+y*x+1,x,0)`.

Reading the documentation or the code first could help...